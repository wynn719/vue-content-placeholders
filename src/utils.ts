export function usePrefix(prefix: string) {
  return `vue-content-placeholders-${prefix}` as const
}
