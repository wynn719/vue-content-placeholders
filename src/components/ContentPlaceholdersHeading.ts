import { defineComponent, h } from 'vue-demi'
import { usePrefix } from '@/utils'

export default defineComponent({
  props: {
    img: {
      type: Boolean,
      default: false,
    },
  },
  setup(props) {
    const className = usePrefix('heading')

    return () =>
      h('div', { class: className }, [
        props.img ? h('div', { class: `${className}__img` }) : null,
        h('div', { class: `${className}__content` }, [
          h('div', { class: `${className}__title` }),
          h('div', { class: `${className}__subtitle` }),
        ]),
      ])
  },
})
