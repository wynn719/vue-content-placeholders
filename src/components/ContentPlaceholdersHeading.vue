<template>
  <div :class="className">
    <div v-if="img" :class="`${className}__img`" />
    <div :class="`${className}__content`">
      <div :class="`${className}__title`" />
      <div :class="`${className}__subtitle`" />
    </div>
  </div>
</template>

<script lang="ts" setup>
import { usePrefix } from '@/utils'

defineProps({
  img: {
    type: Boolean,
    default: false,
  },
})

const className = usePrefix('heading')
</script>
