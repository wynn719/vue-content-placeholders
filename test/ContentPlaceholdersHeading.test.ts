import { describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'
import ContentPlaceholdersHeading from '@/components/ContentPlaceholdersHeading'
import { usePrefix } from '@/utils'

describe('ContentPlaceholdersHeading', () => {
  const createWrapper = (args?: any) => mount(ContentPlaceholdersHeading, {
    ...args,
  })

  it('should render correctly', () => {
    const wrapper = createWrapper()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should support img', () => {
    const wrapper = createWrapper({
      props: {
        img: true,
      },
    })
    expect(wrapper.find(`.${usePrefix('heading__img')}`).exists()).toBe(true)
  })
})
