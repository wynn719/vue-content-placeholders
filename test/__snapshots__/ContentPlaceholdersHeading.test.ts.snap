// Vitest Snapshot v1

exports[`ContentPlaceholdersHeading > should render correctly 1`] = `
"<div class=\\"vue-content-placeholders-heading\\">
  <!---->
  <div class=\\"vue-content-placeholders-heading__content\\">
    <div class=\\"vue-content-placeholders-heading__title\\"></div>
    <div class=\\"vue-content-placeholders-heading__subtitle\\"></div>
  </div>
</div>"
`;
